libraries/chain/webassembly/runtimes/eos-vm-oc/memory.cpp

```diff
@@ -40,7 +40,7 @@ memory::memory(uint64_t sliced_pages) {
    uintptr_t* const intrinsic_jump_table = reinterpret_cast<uintptr_t* const>(zeropage_base - first_intrinsic_offset);
    const intrinsic_map_t& intrinsics = get_intrinsic_map();
    for(const auto& intrinsic : intrinsics)
-      intrinsic_jump_table[-intrinsic.second.ordinal] = (uintptr_t)intrinsic.second.function_ptr;
+      intrinsic_jump_table[-(int)intrinsic.second.ordinal] = (uintptr_t)intrinsic.second.function_ptr;
```

Comment:
This is an interesting one. The error here gets reported somewhat confusingly as `addition of unsigned offset to 0x7aca262015f8 overflowed to 0x7aca26201570`. I believe what is occurring is that `ordinal` is a `size_t` thus `uint64_t`, and negating it is still a `uint64_t`: now a really large `uint64_t`. But pointer arithmetic is defined on integer math; `ptrdiff_t` is a signed integer for example. So I think this is causing some sort of wrap around on a large signed integer. Casting `ordinal` to an `int` before the negation ensures it remains a small value `int` before pointer arithmetic. I'd be curious on other takes though.

Reply:
`ordinal` is the host function index over in,
https://github.com/AntelopeIO/leap/blob/968a0fa0d0aa5c36ea2f2ed3eb4469068a9afd68/libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/intrinsic_mapping.hpp#L18-L19
so `int` will always be plenty large enough (we won't have 2<sup>31</sup> host functions). My initial impression of `intptr_t` is that its purpose/connotation isn't really for indexes in to an array.

I do wonder if `ordinal` should just be an `int` instead; I don't think it'd be too hard to change it.